src/core/server/saved_objects/import/import_saved_objects.ts

```diff
@@ -72,20 +73,35 @@ export async function importSavedObjectsFromStream({
     supportedTypes,
   });
   errorAccumulator = [...errorAccumulator, ...collectSavedObjectsResult.errors];
-  /** Map of all IDs for objects that we are attempting to import; each value is empty by default */
+  // Map of all IDs for objects that we are attempting to import, and any references that are not included in the read stream;
+  // each value is empty by default
   let importStateMap = collectSavedObjectsResult.importStateMap;
   let pendingOverwrites = new Set<string>();
 
+  // Check any references that aren't included in the import file and retries, to see if they have a match with a different origin
+  const checkReferenceOriginsResult = await checkReferenceOrigins({
+    savedObjectsClient,
+    typeRegistry,
+    namespace,
+    importStateMap,
+  });
+  importStateMap = new Map([...importStateMap, ...checkReferenceOriginsResult.importStateMap]);
+
   // Validate references
   const validateReferencesResult = await validateReferences(
     collectSavedObjectsResult.collectedObjects,
     savedObjectsClient,
-    namespace
+    namespace,
+    importStateMap
   );
   errorAccumulator = [...errorAccumulator, ...validateReferencesResult];
 
   if (createNewCopies) {
-    importStateMap = regenerateIds(collectSavedObjectsResult.collectedObjects);
+    importStateMap = new Map([
+      ...importStateMap, // preserve any entries for references that aren't included in collectedObjects
+      ...regenerateIds(collectSavedObjectsResult.collectedObjects),
+    ]);
+    // TODO: check reference origins!
```

Comment:
Is this TODO supposed to be addressed in this PR?

Reply:
Ah that was a note to myself that I added before I started changing code, thanks for catching it.

I wound up checking reference origins before this instead of after it. So I'll remove this TODO.